### DISCUSSION TITLE
Issue with setting metadata in .NET Client (v6.2.0 and v6.0.4)

### DISCUSSION THREAD

**Discussion by andersgranberg** [ravendb/ravendb#19508] [Q&A] [answered]
Hello,

I'm encountering an issue while trying to update the metadata of a document using the RavenDB .NET client (tested with both v6.2.0 and v6.0.4). In the example below, the FooRefresh property on the document itself updates as expected, but the metadata change doesn't persist after SaveChanges(). (Also tried with custom metadata and not just the 'refresh' property)

Here’s the code I'm using:

```using var session = _documentStore.OpenSession();
var document = session.Load<dynamic>([docID]);
document.FooRefresh = DateTime.UtcNow.AddHours(1);
session.Advanced.GetMetadataFor(document)["@refresh"] = DateTime.UtcNow.AddHours(1);
session.SaveChanges();
```

The same operation works as expected when using the Node.js client. I'm wondering if there are any known issues with the .NET client in the 6.X series related to metadata updates, or if I might be missing something in my implementation.

Any advice or guidance would be much appreciated!

Thanks in advance.

**Comment by ppekrol** [comment]
hi @andersgranberg 

```csharp
var document = session.Load<dynamic>([docID]);
```

This line loads `Dictionary<string, dynamic>` because you passed an array of id's to the load method.
You need then to specify for which document you want to modify the metadata:

```csharp
var myDocument = document[docID];
session.Advanced.GetMetadataFor(myDocument)["@refresh"] = DateTime.UtcNow.AddHours(1);
```

**Reply by andersgranberg** [reply]
I'm sorry, I see why you would think that, I didn't meant to display the ID as an array. I'm only loading one id (string) into the Load.

**Reply by andersgranberg** [reply]
@ppekrol Please re-open this discussion

**Reply by ppekrol** [reply]
I reproduced it. It is related to the usage of 'dynamic': https://issues.hibernatingrhinos.com/issue/RavenDB-23055/Cannot-modify-metadata-of-the-document-if-it-was-loaded-using-dynamic-or-object

**Reply by ppekrol** [reply, marked as answer]
The problem is related to the fact that 'dynamic' or 'object' passed as a type to `Load<T>` is treated as a 'raw'. Then we do not do any manipulations on the entity itself, meaning that the '@metadata' from the object is not being removed and during save takes precedence over the session modified one. Workaround for now would be to use a proper type as `T` instead of using 'dynamic'.

**Reply by andersgranberg** [reply]
I understand. Using a typed load works great. Thanks!
